File: samples/theme-preview.go
// Preview script for the QSpad theme
// Comments, keywords, strings, numbers, and functions should all
// read as the same flat color against the ivory background.

package main

import (
	"fmt"
)

const threshold = 0.05

type geneResult struct {
	gene    string
	pValue  float64
}

func (r geneResult) significant() bool {
	return r.pValue < threshold
}

func loadExpressionData(path string, minReads int) ([]geneResult, error) {
	results := []geneResult{
		{"ATP6V0A1", 0.001},
		{"BRCA1", 0.08},
		{"TP53", 0.042},
	}
	return results, nil
}

func summarize(results []geneResult) {
	for _, r := range results {
		status := "not significant"
		if r.significant() {
			status = "significant"
		}
		fmt.Printf("%s: p = %.3f (%s)\n", r.gene, r.pValue, status)
	}
}

func main() {
	results, err := loadExpressionData("./data", 10)
	if err != nil {
		panic(err)
	}
	summarize(results)
}

File: samples/theme-preview.rs
// Preview script for the QSpad theme
// Comments, keywords, strings, numbers, and functions should all
// read as the same flat color against the ivory background.

const THRESHOLD: f64 = 0.05;

struct GeneResult {
    gene: String,
    p_value: f64,
}

impl GeneResult {
    fn significant(&self) -> bool {
        self.p_value < THRESHOLD
    }
}

fn load_expression_data() -> Vec<GeneResult> {
    vec![
        GeneResult { gene: "ATP6V0A1".to_string(), p_value: 0.001 },
        GeneResult { gene: "BRCA1".to_string(), p_value: 0.08 },
        GeneResult { gene: "TP53".to_string(), p_value: 0.042 },
    ]
}

fn summarize(results: &[GeneResult]) {
    for r in results {
        let status = if r.significant() { "significant" } else { "not significant" };
        println!("{}: p = {:.3} ({})", r.gene, r.p_value, status);
    }
}

fn main() {
    let results = load_expression_data();
    summarize(&results);
}
